### PR TITLE
Logs are created dynamically based on network name - Closes #2452

### LINKED ISSUE #2452: Create log file dynamically based on network
### Expected behavior
Logfile name should be created dynamically based on the network if I create a network `testnet-sync` then the log file should be `lisk-testnet-sync.log` or we can create separate log directory specific to network like `logs/testnet-sync/lisk.log`, either of the approaches should be good.

This way if a user is running multiple node from same code base at same place just by having custom network configuration then the logs will be created dynamically based on the network rather than changing the log file name in the custom network config file.

### Actual behavior
Currently, when we start a network with parameter `testnet-sync`, we need to specify explicitly the log file name for custom log file name.

### PR DESCRIPTION
**Pull request by pablitovicente**
### What was the problem?

Logs were not unique to each network.

### How did I fix it?

Based on PR feedback I added the full path to each network. When creating a custom network the paths need to be set on its config file.

### How to test it?

- Run  `node app.js` and it should create `logs/devnet` (default network name)
- Create a config, set the log paths for both properties logFile and db.logFile and run `node app.js -n <your_network_name>` and it should create the logs in `logs/<your_network_name> ` 

### Review checklist

* The PR resolves #2452
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -159,6 +159,7 @@ var logger = new Logger({
 	echo: process.env.LOG_LEVEL || appConfig.consoleLogLevel,
 	errorLevel: process.env.FILE_LOG_LEVEL || appConfig.fileLogLevel,
 	filename: appConfig.logFileName,
+	network: appConfig.network,
```
Comment (nazarhussain) [inline app.js]: Suggest not to add network specific dependency to logger. It's against of the separation of concern rule. Logger just need to log, for it it needs a file path, log level and any additional config attribute for logging. 

The logic to choose the log file path based on network is responsibility of one level higher.

File: logger.js [hunk 2]
```diff
@@ -51,8 +51,16 @@ module.exports = function(config) {
 
 	config.errorLevel = config.errorLevel || 'log';
 
-	child_process.execSync(`mkdir -p ${path.dirname(config.filename)}`);
-	var log_file = fs.createWriteStream(config.filename, { flags: 'a' });
+	const logsSubDirectory = config.network || '';
+	const logsDirectory = path.join(
+		path.dirname(config.filename),
+		logsSubDirectory
```
Comment (nazarhussain) [inline logger.js]: Here it makes things confusing. The log file path is a node configuration option. User set it to some path, and you changed that path runtime, make the user wondering where his log file is. 

So right way to do is to just add a custom configuration override for the network with different log file paths.

**Comment by nazarhussain** [comment]
@Nazgolze @fchavant This change requires your attention to relevant change in archiving, logrotate or graylogs. 

@MaciejBaj We should highlight this change promptly, as user can get confused finding their logs after this change.

**Comment by ManuGowda** [comment]
@nazarhussain the DevOps team is informed already and as part of the dev updates we should cover this and same can be communicated in the chat or wherever necessary.

**Pull request merged**